### ISSUE TITLE
Kubectl exec failed with internal server error message

### ISSUE THREAD

**Issue by galal-hussein** [rancher/rancher#4664]
#### Rancher Version:
v1.1.0-dev1-rc2
#### Docker Version:
1.10.3
#### OS: 
Ubuntu 14.04
#### Steps to Reproduce: 
Create any pod and try to run kubectl exec from the kubectl in UI:

```
> kubectl exec busybox-sleep  -- ls /
error: unexpected response status code 500 (Internal Server Error)
```

Running with verbosity:

```
> kubectl exec --v=9 busybox-sleep -c busybox -- ls /
I0503 19:12:26.202275     272 debugging.go:102] curl -k -v -XGET  -H "User-Agent: kubectl/v1.1.7 (linux/amd64) kubernete
s/e4e6878" http://master/api
I0503 19:12:26.207766     272 debugging.go:121] GET http://master/api 200 OK in 5 milliseconds
I0503 19:12:26.207839     272 debugging.go:127] Response Headers:
I0503 19:12:26.207853     272 debugging.go:130]     Content-Type: application/json
I0503 19:12:26.207874     272 debugging.go:130]     Date: Tue, 03 May 2016 19:12:26 GMT
I0503 19:12:26.207909     272 debugging.go:130]     Content-Length: 132
I0503 19:12:26.207968     272 request.go:746] Response Body: {"kind":"APIVersions","versions":["v1"],"serverAddressByCli
entCIDRs":[{"clientCIDR":"0.0.0.0/0","serverAddress":"172.17.0.4:443"}]}
I0503 19:12:26.208591     272 debugging.go:102] curl -k -v -XGET  -H "User-Agent: kubectl/v1.1.7 (linux/amd64) kubernete
s/e4e6878" http://master/api/v1/namespaces/default/pods/busybox-sleep
I0503 19:12:26.211094     272 debugging.go:121] GET http://master/api/v1/namespaces/default/pods/busybox-sleep 200 OK in
 2 milliseconds
I0503 19:12:26.211179     272 debugging.go:127] Response Headers:
I0503 19:12:26.211205     272 debugging.go:130]     Content-Length: 1146
I0503 19:12:26.211239     272 debugging.go:130]     Content-Type: application/json
I0503 19:12:26.211282     272 debugging.go:130]     Date: Tue, 03 May 2016 19:12:26 GMT
I0503 19:12:26.211449     272 request.go:746] Response Body: {"kind":"Pod","apiVersion":"v1","metadata":{"name":"busybox
-sleep","namespace":"default","selfLink":"/api/v1/namespaces/default/pods/busybox-sleep","uid":"ccf1df8e-1160-11e6-983e-
02dffdf7cbbc","resourceVersion":"8124","creationTimestamp":"2016-05-03T18:56:53Z"},"spec":{"containers":[{"name":"busybo
x","image":"busybox","args":["sleep","1000000"],"resources":{},"terminationMessagePath":"/dev/termination-log","imagePul
lPolicy":"IfNotPresent"}],"restartPolicy":"Always","terminationGracePeriodSeconds":30,"dnsPolicy":"ClusterFirst","nodeNa
me":"node-02","securityContext":{}},"status":{"phase":"Running","conditions":[{"type":"Ready","status":"True","lastProbe
Time":null,"lastTransitionTime":"2016-05-03T18:56:56Z"}],"hostIP":"139.59.169.227","podIP":"10.42.164.66","startTime":"2
016-05-03T18:56:53Z","containerStatuses":[{"name":"busybox","state":{"running":{"startedAt":"2016-05-03T18:56:55Z"}},"la
stState":{},"ready":true,"restartCount":0,"image":"busybox","imageID":"docker://sha256:47bcc53f74dc94b1920f0b34f60360965
26296767650f223433fe65c35f149eb","containerID":"docker://ce6d6f02ee4cf0092952b394d311fa5aff8f62ccd68ba015cdc8368b4238312
2"}]}}
I0503 19:12:26.212849     272 debugging.go:102] curl -k -v -XPOST  -H "X-Stream-Protocol-Version: v2.channel.k8s.io" -H
"X-Stream-Protocol-Version: channel.k8s.io" http://master/api/v1/namespaces/default/pods/busybox-sleep/exec?command=ls&c
ommand=%2F&container=busybox&container=busybox&stderr=true&stdout=true
I0503 19:12:26.233877     272 debugging.go:121] POST http://master/api/v1/namespaces/default/pods/busybox-sleep/exec?com
mand=ls&command=%2F&container=busybox&container=busybox&stderr=true&stdout=true 500 Internal Server Error in 20 millisec
onds
I0503 19:12:26.233999     272 debugging.go:127] Response Headers:
I0503 19:12:26.234026     272 debugging.go:130]     Content-Type: application/json
I0503 19:12:26.234066     272 debugging.go:130]     Date: Tue, 03 May 2016 19:12:26 GMT
I0503 19:12:26.234090     272 debugging.go:130]     Content-Length: 131
F0503 19:12:26.234204     272 helpers.go:96] error: unexpected response status code 500 (Internal Server Error)

```

**Comment by galal-hussein** [comment]
Kubectl attach generate the same error:

```
> kubectl attach  busybox-sleep -c busybox
error: unexpected response status code 500 (Internal Server Error)
```

**Comment by alena1108** [comment]
@galal-hussein what is the version of kubectl service?

**Comment by galal-hussein** [comment]
@alena1108 The client version is 1.2.0 
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"2", GitVersion:"v1.2.0", GitCommit:"5cb86ee022267586db386f62781338b0483733b3", GitTreeState:"clean"}
```
The kubectl in the UI fails too.

**Comment by alena1108** [comment]
https://github.com/kubernetes/kubernetes/issues/22063

**Comment by galal-hussein** [comment]
Kubectl exec/logs/attach tested and worked on the following setup:

rancher/server:v1.2.0-pre1-rc1
rancher/k8s:v1.3.0-rancher1

However it failed on:

rancher/server:v1.2.0-pre1-rc1
rancher/k8s:v1.2.4-rancher9

```
$ kubectl logs nginx-1jjhu
Error from server: Get https://test2:10250/containerLogs/default/nginx-1jjhu/nginx: dial tcp: lookup test2 on 169.254.169.250:53: no such host
$ kubectl exec nginx-1jjhu ls
Error from server: dial tcp: lookup test2 on 169.254.169.250:53: no such host
$ kubectl attach nginx-1jjhu
Error from server: dial tcp: lookup test2 on 169.254.169.250:53: no such host
```

**Comment by sangeethah** [comment]
Reopening this issue , to track the issue seen with logs/exec/attach broken with kubernetes 1.2 .

**Comment by alena1108** [comment]
@sangeethah based on discussion in engineering, all new features/bug fixes will go to 1.3 only. Unless they are some blocker regression bugs.

**Comment by sangeethah** [comment]
Closing this issue with validation results from k8s 1.3 setup using rancher/k8s:v1.3.0-rancher1
